### PR TITLE
replace usages of ``copy_arrays`` with ``memmap``

### PR DESCRIPTION
**Pull request by zacharyburnett**
## PR Description

follow-on to https://github.com/asdf-format/asdf/pull/1797

### CODE REVIEW

File: changelog/7737.trivial.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+replace usages of ``copy_arrays`` with ``memmap``
```
Comment (nabobalis) [inline changelog/7737.trivial.rst:1]: Since this in the test file, this does not need a changelog

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks for the PR @zacharyburnett

**Comment by zacharyburnett** [comment]
@nabobalis I apologize, I forgot to include that `memmap` only exists in `asdf>=3.1.0`; there should also probably be a pin in `pyproject.toml` to that effect, or alternatively I can add code to check the `asdf` version and use `copy_arrays` or `memmap` accordingly

**Comment by nabobalis** [comment]
As long as our test suite passed, which I think it did even on our oldest deps, I don't see the issue?

**Comment by nabobalis** [comment]
The oldest deps despite the pin, used asdf==3.3.0, I assume maybe a 310 cant install older versions?

**Comment by zacharyburnett** [comment]
it looks like `oldestdeps` in `tox.ini` pins `asdf<2.12.0`
https://github.com/sunpy/sunpy/blob/e5878584d57ed21916b390987cfe7d641387540c/tox.ini#L61
however the project requires `asdf>=2.13.0`
https://github.com/sunpy/sunpy/blob/e5878584d57ed21916b390987cfe7d641387540c/pyproject.toml#L58
and since the `pip install .` step occurs after the dependencies from `toxenv:deps` are installed, this means the oldestdeps in `tox.ini` are overwritten:
https://github.com/sunpy/sunpy/actions/runs/9993966424/job/27622904664#step:10:21

**Comment by nabobalis** [comment]
Wait, our CI wasn't updated to use the new min versions?!

I will have to change that.  

Either way, this is just test changes so I have no issue as long as our tests pass.

**Comment by zacharyburnett** [comment]
We could use something like https://github.com/OpenAstronomy/minimum_dependencies to gather the oldest deps and install those (see https://github.com/spacetelescope/jwst/blob/de39de07b6f0380860caa86e8485666b4cef35c2/tox.ini#L51-L64)

**Comment by nabobalis** [comment]
We already do elsewhere, it just wasn't added to sunpy for some reason I don't know.

**Pull request merged**